exporter.go

```diff
@@ -62,11 +63,44 @@ func getEnvDuration(key string, defaultValue time.Duration) time.Duration {
 	return d
 }
 
-func createClientWithRetries(getClient func() (interface{}, error), retries int, retryInterval time.Duration) (interface{}, error) {
+// positiveDuration is a wrapper of time.Duration to ensure only positive values are accepted
+type positiveDuration struct{ duration *time.Duration }
+
+func (pd positiveDuration) String() string {
+	return fmt.Sprint(pd.duration)
+}
+
+func (pd positiveDuration) Set(s string) error {
+	if pd.isNegative(s) {
+		return fmt.Errorf("Negative duration is not valid")
+	}
+	dur, err := time.ParseDuration(s)
+	if err != nil {
+		return err
+	}
+
+	*pd.duration = dur
+	return nil
+}
+
+func (pd positiveDuration) isNegative(s string) bool {
```

Comment:
this should not be a method of positiveDuration.  I'm ok if you use `if s[0] == '-'` in the functions that use `isNegative` now

Reply:
seems like the update to this comment wasn't saved :(

better to create a function `func parsePositiveDuration(string) (positiveDuration, error)` that can be reused and unit tested